samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala

```diff
@@ -245,14 +263,12 @@ class OffsetManager(
 
       // invoke checkpoint listeners
       //partitionOffsets.groupBy(_._1.getSystem).foreach {
-      partitionOffsets.groupBy { case (ssp, _) => ssp.getSystem }.foreach {
+      checkpoint.getOffsets.asScala.groupBy { case (ssp, _) => ssp.getSystem }.foreach {
         case (systemName:String, offsets: Map[SystemStreamPartition, String]) => {
```

Comment:
scala.MatchError: (test-system2,Map(SystemStreamPartition [test-system2, test-stream2, 0] -> 100)) (of class scala.Tuple2)
        at org.apache.samza.checkpoint.OffsetManager$$anonfun$writeCheckpoint$4.apply(OffsetManager.scala:264)